Fix docker stats show wrong memory limit when do docker update

coolljt0725 opened the pull request:
When a container create with -m 100m and then docker update other
cgroup settings such as --cpu-quota, the memory limit show by
docker stats will become the default value but not the 100m.

Signed-off-by: Lei Jitang leijitang@huawei.com

Reproduce:
`docker run -d -ti -m 100m --name bar busybox`
`docker stats bar`

<pre><code>CONTAINER           CPU %               MEM USAGE / LIMIT     MEM %               NET I/O             BLOCK I/O
bar                 0.00%               401.4 kB / 104.9 MB   0.38%               648 B / 648 B       0 B / 0 B
</code></pre>

As we can see the MEM LIMIT is 104.9MB just what we set on run
`docker update --cpu-quota 2000 bar`

<pre><code>CONTAINER           CPU %               MEM USAGE / LIMIT     MEM %               NET I/O             BLOCK I/O
bar                 0.00%               421.9 kB / 4.144 GB   0.01%               648 B / 648 B       0 B / 0 B
</code></pre>

The `MEM LIMIT` become the default value.

Has discussed with @hqhq , this should be the correct fix.

icecrime commented:
Related?

```
----------------------------------------------------------------------
FAIL: docker_cli_update_unix_test.go:167: DockerSuite.TestUpdateStats

docker_cli_update_unix_test.go:174:
    c.Assert(waitRun(id), checker.IsNil)
... value *exec.ExitError = &exec.ExitError{ProcessState:(*os.ProcessState)(0xc8204efa00)} ("exit status 1")
```

icecrime commented:
Otherwise code LGTM, thanks!

coolljt0725 committed:
Fix docker stats show wrong memory limit when do docker update

When a container create with -m 100m and then docker update other
cgroup settings such as --cpu-quota, the memory limit show by
docker stats will become the default value but not the 100m.

Signed-off-by: Lei Jitang <leijitang@huawei.com>

coolljt0725 commented:
@icecrime  test pass, windows failure is not related

cpuguy83 commented:
LGTM

thaJeztah commented:
looks like we have two LGTM's, so moving to "merge"

I'll kick windows build again for good measures.

calavera commented:
LGTM
